scripts/evaluator.py

```diff
@@ -0,0 +1,338 @@
+
+"""
+Evaluate classification and (optionally) face detection ability on a set of videos.
+Videos to be evaluated should be from the TestVideos folder on the Drive.
+To compare face detection models, run annotator.py first.
+"""
+
+
+class Evaluator():
+    def __init__(self, cuda, detector, detector_type, classifier, input_directory, rate, det_file):
+        """
+        Evaluates face detection and goggle classification performance.
+        Goggle Classification accuracy is given by average class accuracy and individual
+        video accuracy.
+        Face detection accuracy is given by precision and recall values.
+
+        @param cuda: A bool value that specifies if cuda shall be used
+        @param detector: A string path to a .pth weights file for a face detection model
+        @param detector_type: One of 'blazeface', 'ssd', 'retinaface'.
+        @param classifier: A string path to a .pth weights file for a goggle classification model
+        @param input_directory: Directory containing test videos to run Evaluator on
+        @param rate: Run detection and classification on every 1/rate frames
+        @param det_file: CSV generated by annotator.py containing detection results
+        """
+
+        if cuda and torch.cuda.is_available():
+            torch.set_default_tensor_type('torch.cuda.FloatTensor')
+            self.device = torch.device('cuda:0')
+        else:
+            torch.set_default_tensor_type('torch.FloatTensor')
+            self.device = torch.device('cpu')
+
+        if os.path.exists(PRED_DETECTIONS_FILE):
+            os.remove(PRED_DETECTIONS_FILE)
+
+        self.detector = FaceDetector(detector=detector, detector_type=detector_type,
+                                     cuda=cuda and torch.cuda.is_available(), set_default_dev=True)
+
+        weights = torch.load(classifier, map_location=self.device)
+        if isinstance(weights, dict):
+            # if the .pth is just a state_dict, we need to
+            # load the model from goggle_classifier.py
+            model = get_model()
+            model.load_state_dict(weights)
+            weights = model
+
+        self.classifier = Classifier(weights, cuda)
+        self.video_filenames = self.get_video_files(input_directory)
+        self.results = {'Goggles':
+                            {'average_class_accuracy': 0.0,
+                             'number_of_videos': 0,
+                             'individual_video_results': {}
+                             },
+                        'Glasses':
+                            {'average_class_accuracy': 0.0,
+                             'number_of_videos': 0,
+                             'individual_video_results': {}
+                             },
+                        'Neither':
+                            {'average_class_accuracy': 0.0,
+                             'number_of_videos': 0,
+                             'individual_video_results': {}
+                             }
```

Comment:
this is a really large dictionary. especially with its usage in `record_results` I think it might be better suited as its own object

Reply:
Yeah this dictionary gets added to throughout the script and then saved as a json file with detailed results for each video that was evaluated

Reply:
If I make this its own class, would I create methods for updating the different values in it?